### ISSUE TITLE
'openshift config' command rejects all patches as non valid JSON

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#493]
```
Environment
OS: Windows 7
Minishift version: minishift 1.0.0-beta.4-1, cdk 3.0.0-beta.2
```
I am following docs - [Updating OpenShift configuration](https://github.com/minishift/minishift/blob/724f750d5b1116b8651d5b5f75378906ed91a357/docs/using.md#updating-openshift-configuration). Running example command in both cmd and powershell gives me following message: "The specified patch need to be a valid JSON." I tried several alternations of quotes but without success:

```
PS C:\Users\CDK> minishift openshift config set --patch '{"corsAllowedOrigins": [".*"]}'
The specified patch need to be valid JSON.

PS C:\Users\CDK> minishift openshift config set --patch "{'corsAllowedOrigins': ['.*']}"
The specified patch need to be valid JSON.

PS C:\Users\CDK> minishift openshift config set --patch '{apiVersion : v1}'
The specified patch need to be valid JSON.
         
C:\Users\CDK\.minishift\cache\oc\v3.4.1.2>minishift openshift config set --patch "{apiVersion:v1}"
The specified patch need to be valid JSON.
     
C:\Users\CDK\.minishift\cache\oc\v3.4.1.2>minishift openshift config set --patch "{ apiVersion:v1 }"
The specified patch need to be valid JSON.
     
C:\Users\CDK\.minishift\cache\oc\v3.4.1.2>minishift openshift config set --patch "{ apiVersion : v1 }"
The specified patch need to be valid JSON.
```

**Comment by hferentschik** [comment]
@agajdosi Have you tried wrapping the string in '@', using here strings - http://stackoverflow.com/questions/18116186/escaping-quotes-and-double-quotes?

**Comment by agajdosi** [comment]
@hferentschik Strangely it seems my power shell v2.0 does not recognize herestrings.

    PS C:\Users\CDK> minishift openshift config set --patch @'{"corsAllowedOrigins" : [".*"]}'@
    Unrecognized token in source text.
    At line:1 char:40
    + minishift openshift config set --patch  <<<< @'{"corsAllowedOrigins" : [".*"]}'@
        + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
        + FullyQualifiedErrorId : UnrecognizedToken
     
Cmd however works well with it, but it does not help with issue:

    C:\Users\CDK>minishift openshift config set --patch @'{"corsAllowedOrigins" : [".*"]}'@
    The specified patch need to be valid JSON.
     
This lead me to that this is not about string but about what is parsing JSON:

    PS C:\Users\CDK> $command='{"corsAllowedOrigins" : [".*"]}'
    PS C:\Users\CDK> echo $command
    {"corsAllowedOrigins" : [".*"]}

**Comment by budhrg** [comment]
Hi @agajdosi ,

This is interesting. I searched a bit around this and found it is common thing in powershell.

Try https://msdn.microsoft.com/en-us/powershell/reference/3.0/microsoft.powershell.utility/convertto-json
or http://stackoverflow.com/questions/33217907/powershell-convert-string-to-json .

We need to properly document it how we get it correct.

**Comment by hferentschik** [comment]
This is what works for me:

    $  .\minishift.exe --% openshift config set --patch "{\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}"

See http://stackoverflow.com/questions/24745868/powershell-passing-json-string-to-curl

So you need '--%' to stop powershell to do any kind of substitutions and then you need to escape the " themselves. Not pretty, I agree. Moving forward we will offer a variant of this where you can point to a file. This way you don't have to deal with escaping. 

That said, think about this way.  {"routingConfig": {"subdomain": "192.168.99.101.nip.io"}} is the JSON. You need to make sure that exactly this gets into Minishift. Depending on your OS and shell type you will have to think about how to do that. 

@agajdosi, can you confirm this works? We probably should add this to the docs. At least the Powershell version is important enough.

**Comment by agajdosi** [comment]
@hferentschik I have been experimenting with --% but I do not remember this particular tip with escaping " with \, so it is possible it will help. However I do not have Windows environment here right now, so confirmation will come probably on monday.

**Comment by agajdosi** [comment]
@hferentschik Found that --% flag is not implemented in Powershell v2 which is native on Windows 7. However tried to play with your proposed solution and got it to work.

Workaround for CMD is easy:

    minishift.exe openshift config set --patch "{\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}"

For Powershell v2 I've found that it does not consume it directly, however when putted in via variable then it is all fine:

    PS C:\Users\CDK\minishift> $json='{\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}'
    PS C:\Users\CDK\minishift> echo $jason
    {\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}
    PS C:\Users\CDK\minishift> minishift.exe openshift config set --patch $jason
    Patching OpenShift configuration /var/lib/origin/openshift.local.config/master/master-config.yaml with {"routingConfig": {"subdomain": "192.168.99.101.nip.io"}}
    Restarting OpenShift

I've checked results for both with 'minishift openshift config view'.

**Comment by hferentschik** [comment]
@agajdosi, cool. So for now really a case of documentation I guess. WDYT?

**Comment by agajdosi** [comment]
@hferentschik I agree, for me it looks like problem of command lines on Windows 7, good to cover it in documentation.

**Comment by agajdosi** [comment]
@Preeticp Situation is exactly the same on Windows 10 as it is on Windows 7.
For cmd:
`minishift.exe openshift config set --patch "{\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}"`
For powershell workaround with variable:
`$json='{\"routingConfig\": {\"subdomain\": \"192.168.99.101.nip.io\"}}' `

**Comment by hferentschik** [comment]
Merged via pull request #656
